### PR TITLE
Feat/classes list

### PR DESCRIPTION
**Pull request by komalahire**
Made List of classes feature and used Modal too.

### CODE REVIEW

File: src/components/Class/ClassesList/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            <th>Description</th>
+            <th>Language</th>
+            <th>Class type</th>
+            <th>Date</th>
+            <th> Start Time </th>
+            <th>End Time</th>
+          </tr>
+        </thead>
+        {allClass.map((item, index) => {
+          return (
```
Comment (vikotar) [inline src/components/Class/ClassesList/index.js]: Could we move to its separate component? The header is fine. the item/class I think we can move into different component.

File: src/components/Class/ClassesList/styles.scss [hunk 2]
```diff
@@ -0,0 +1,110 @@
+
+  
+
+.createClassButton{
```
Comment (vikotar) [inline src/components/Class/ClassesList/styles.scss]: css naming convention usually goes like `create-class-button`.  What do you think if we use this kind of class names in css files? 🤔

File: src/components/Class/ClassesList/styles.scss [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	border-radius: 15px;
+	color: white;
+	font-size: 17px;
+	font-weight:'bold';
+	border: none ;
+}
+
+
+
+//  style={{marginLeft:'17%',backgroundColor:'#83B344',width:'10%',height:'40px',borderRadius:'15px',color:'white',fontSize:'17px',fontWeight:'Bold',border:'none'}}
```
Comment (vikotar) [inline src/components/Class/ClassesList/styles.scss]: If we don't need it. We should remove the commented code.

File: src/components/Class/index.js [hunk 4]
```diff
@@ -44,7 +44,7 @@ function Class() {
   const { user : { rolesList = [] } } = useSelector(({User}) => User.data)
```
Comment (vikotar) [inline src/components/Class/index.js]: I think we can remove the logic from line number 44 to 49. We are now giving access to create a class to anyone. cc @pysaquib

File: src/pages/CreateClass/index.js [hunk 5]
```diff
@@ -1,10 +1,37 @@
...
 import CreateClassComponent from '../../components/Class'
+import ClassesList from '../../components/Class/ClassesList'
+import Modal from '../../components/common/Modal'
+import "../../components/Class/ClassesList/styles.scss"
+
 
 function CreateClass() {
+  const [open, setOpen] = React.useState(false);
+
+  const handleClickOpen = () => {
```
Comment (vikotar) [inline src/pages/CreateClass/index.js:11]: In my view the better name could be `toggleClassFormModal`.  what do you think?

File: src/pages/CreateClass/index.js [hunk 6]
```diff
@@ -1,10 +1,37 @@
 import React from "react";
-
 import CreateClassComponent from '../../components/Class'
+import ClassesList from '../../components/Class/ClassesList'
+import Modal from '../../components/common/Modal'
+import "../../components/Class/ClassesList/styles.scss"
+
 
 function CreateClass() {
+  const [open, setOpen] = React.useState(false);
```
Comment (vikotar) [inline src/pages/CreateClass/index.js]: Instead of `open`, We can use `[ showModel, setShowModel]`? which is more descriptive.

**Review by vikotar — APPROVED** [review]

File: src/components/Class/ClassesList/styles.scss (hunk 2 above)
Comment (komalahire) [reply src/components/Class/ClassesList/styles.scss]: Done

File: src/components/Class/index.js (hunk 4 above)
Comment (komalahire) [reply src/components/Class/index.js]: Done

File: src/pages/CreateClass/index.js (hunk 6 above)
Comment (komalahire) [reply src/pages/CreateClass/index.js]: done

File: src/components/Class/ClassesList/index.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
+import React, { useEffect } from "react";
+import { useSelector, useDispatch } from 'react-redux';
```
[the remark this answers is not in the document]
Comment (komalahire) [reply src/components/Class/ClassesList/index.js]: yes, done

File: src/components/Class/ClassesList/styles.scss (hunk 3 above)
Comment (komalahire) [reply src/components/Class/ClassesList/styles.scss]: Done

File: src/pages/CreateClass/index.js (hunk 5 above)
Comment (komalahire) [reply src/pages/CreateClass/index.js:11]: Done

File: src/components/Class/ClassList/index.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
+import React, { useEffect } from "react";
+import { useSelector, useDispatch } from 'react-redux';
+import moment from 'moment';
+import { actions as classActions } from '../redux/action'
+import Loader from '../../common/Loader'
+import "./styles.scss";
+function ClassList() {
```
Comment (vikotar) [inline src/components/Class/ClassList/index.js:11]: Could we put 2 blank lines before the component starts?

File: src/components/Class/ClassList/index.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
+import React, { useEffect } from "react";
+import { useSelector, useDispatch } from 'react-redux';
+import moment from 'moment';
```
Comment (vikotar) [inline src/components/Class/ClassList/index.js:4]: Could we put one blank line before our file imports? Somthing like this. 
```
import React, { useEffect } from "react";
import { useSelector, useDispatch } from 'react-redux';
import moment from 'moment';
// one blank line after package imports
import { actions as classActions } from '../redux/action'
import Loader from '../../common/Loader'
import "./styles.scss";

// 2 blank lines before our component code starts.
function ClassList() {
```

File: src/components/Class/ClassList/styles.scss [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+	font-size: 19px;
+	font-family: "Times New Roman", Times, serif;
+	}
+
+
+@media 
+only screen and (max-width: 760px),
+(min-device-width: 768px) and (max-device-width: 1024px)  {
+
+	table {
```
Comment (vikotar) [inline src/components/Class/ClassList/styles.scss]: Could we fix the indentation? You can take a look in other `styles.scss` files. 🙂

File: src/components/Class/ClassList/index.js (hunk 8 above)
Comment (komalahire) [reply src/components/Class/ClassList/index.js:11]: done

File: src/components/Class/ClassList/index.js (hunk 9 above)
Comment (komalahire) [reply src/components/Class/ClassList/index.js:4]: done

File: src/constant.js [hunk 11]
```diff
@@ -6,4 +6,5 @@ export const PATHS = {
   CLASS: "/class",
   COURSE: "/course",
   COURSE_CONTENT: "/course/:courseId",
+  CLASSLIST: "/classList",
```
Comment (vikotar) [inline src/constant.js]: I think we can remove the `CLASSLIST`. We might not need it.

File: src/routing/index.js [hunk 12]
```diff
@@ -27,6 +27,8 @@ const Routing = () => {
         component={ Class }
       />*/}
       <PrivateRoute exact path={PATHS.CLASS} component={CreateClass} />
+
+      <PrivateRoute exact path={PATHS.CLASSLIST} component={ClassList} />
```
Comment (vikotar) [inline src/routing/index.js]: We can also remove this `<PrivateRoute exact path={PATHS.CLASSLIST} component={ClassList} />`.

File: src/routing/index.js [hunk 13]
```diff
@@ -1,14 +1,14 @@
...
 import { Route } from "react-router-dom";
 import { PrivateRoute } from "./Routes";
 // import { userRoles } from '../services/auth'
 import { PATHS } from "../constant";
-
 import Login from "../pages/Login";
 import CreateClass from "../pages/CreateClass";
 import Course from "../pages/Course";
 import CourseContent from "../pages/CourseContent";
+import ClassList from "../components/Class/ClassList";
```
Comment (vikotar) [inline src/routing/index.js]: We can also remove this. `import ClassList from "../components/Class/ClassList";`

File: src/components/Class/ClassList/styles.scss (hunk 10 above)
Comment (komalahire) [reply src/components/Class/ClassList/styles.scss]: done

File: src/constant.js (hunk 11 above)
Comment (komalahire) [reply src/constant.js]: okay done

File: src/routing/index.js (hunk 12 above)
Comment (komalahire) [reply src/routing/index.js]: done

File: src/routing/index.js (hunk 13 above)
Comment (komalahire) [reply src/routing/index.js]: done

**Pull request merged**